__tests__/actions/executeFetch.spec.js

```diff
@@ -130,6 +130,49 @@ describe('executeFetch', () => {
       const data = await thunk(dispatch, getState);
       expect(data).toEqual([{ id: '123', name: 'joe' }]);
     });
+
+    it('should use specified REST method in options if provided', async () => {
+      Object.assign(config, {
+        defaultOpts: { default: 'opt' },
+        resources: {
+          users: {
+            fetch: () => ({
+              url: 'http://api.domain.com/users/:id',
+              opts: {
+                resource: 'opt',
+              },
+            }),
+          },
+        },
+      });
+      fetch.mockResponseOnce(
+        JSON.stringify([{ id: '123', name: 'joe' }, { id: '456', name: 'josephine' }]),
+        { status: 200, headers: { 'Content-Type': 'application/json' } }
+      );
+      const thunk = executeFetch({
+        resource, id, opts: { ...opts, method: 'PUT' }, actionType: 'UPDATE_COLLECTION',
+      });
+      const data = await thunk(dispatch, getState);
+      expect(data).toEqual([{ id: '123', name: 'joe' }, { id: '456', name: 'josephine' }]);
+      expect(fetch).toHaveBeenCalledWith(
+        'http://api.domain.com/users/123',
+        {
+          default: 'opt', method: 'PUT', resource: 'opt', some: 'opt',
+        }
+      );
+      const promise = Promise.resolve(data);
+      expect(dispatch).toHaveBeenCalledWith({
+        type: types.UPDATE_COLLECTION_STARTED,
+        resource,
+        id,
+        opts: {
+          ...opts,
+          method: 'PUT',
```

Comment:
Do we want this override to exist for all action creators? Can we confine this override to this specific action creator chain `UPDATE_COLLECTION`?

Reply:
Would it be better to allow for the overriding of `method` across all action creators but perhaps in the README mention a word of caution if this is being overwritten, you have to know what you're doing? This allows for more flexibility in this library.

Reply:
The `iguazu-rest` library is an opinionated REST library. If developers wish a more flexible unopinionated library, they can use `iguazu-rpc`.